service/src/LevelZero.cpp

```diff
@@ -178,7 +181,18 @@ namespace geopm
             perf_domain_cache(gpu_idx);
             engine_domain_cache(gpu_idx);
             temperature_domain_cache(gpu_idx);
-       }
+            metric_group_init(gpu_idx);
+        }
+    }
+
+    LevelZeroImp::~LevelZeroImp() {
+        for (unsigned int gpu_idx = 0; gpu_idx < m_num_gpu; gpu_idx++) {
```

Comment:
```suggestion
        for (unsigned int gpu_idx = 0; gpu_idx < m_num_gpu; ++gpu_idx) {
```

Please be consistent with the variable incrementing.